Launchpad to Proto Migration for platform alignment, tank cleaning and DMI

siewqin-structo opened the pull request.

rcyl-structo commented on dtf/plc.py:
```diff
@@ -300,6 +356,36 @@ def trigger(self, device, on):
+            print("Time left: {} min {} s".format(timeLeft//60, timeLeft%60))
+            time.sleep(1)
+            if elapsed.seconds >= timeSec:
+                self.send(cmds["light"] + "0")
```
change to send_command to be consistent

siewqin-structo replied:
Corrected

rcyl-structo commented on dtf/plc.py:
```diff
@@ -300,6 +356,36 @@ def trigger(self, device, on):
+            if elapsed.seconds >= timeSec:
+                self.send(cmds["light"] + "0")
+                print("LCD Off")
+                self.send(cmds["fan"] + "0")
```
change to send_command to be consistent

siewqin-structo replied:
Corrected

rcyl-structo commented on dtf/plc.py:
```diff
@@ -325,51 +411,43 @@ def print_layer(self, layer_thickness, exp_time_ms, pre_set_time_ms, post_set_ti
+        # Question: check door bit here?
 
+        self.send_param(layer_thickness, exp_time_ms, pre_set_time_ms, post_set_time_ms, \
+        post_reflow_time_ms, separation_phase1_dist, separation_phase2_dist, \
+        return_phase2_dist, separation_phase1_speed, separation_phase2_speed, \
+        return_phase1_speed, return_phase2_speed, flag)
```
formatting does not look good here, try to align the parameter list with the first argument

siewqin-structo replied:
Corrected

rcyl-structo commented on dtf/dtfserver.py:
```diff
@@ -25,19 +25,119 @@
+            return dtfserver_pb2.GenericResponse()
+        print("Raising Platform")
+
+        if (self.plc.poll_status_with_door(self.plc.remove_print_time_sec, self.plc.sequence_complete,1) is False):
```
correct formating

siewqin-structo replied:
Corrected

rcyl-structo commented on dtf/dtfserver.py:
```diff
@@ -25,19 +25,119 @@
+        self.plc.send_command("PLCClearSequenceComplete")
+        if self.plc.check_bit(self.plc.remove_bit, 0):
+            print("HomeVerticalStage Complete")
+            return dtfserver_pb2.GenericResponse(success=True, errorCode=0,  message="HomeVerticalStage complete")
```
correct formatting

siewqin-structo replied:
Corrected

rcyl-structo commented on dtf/dtfserver.py:
```diff
@@ -25,19 +25,119 @@
+                self.handleError("Invalid Op", context)
+                return dtfserver_pb2.GenericResponse()
+
+        if (self.plc.poll_status_with_door(self.plc.remove_print_time_sec, self.plc.sequence_complete, 1) is False):
```
correct formatting

siewqin-structo replied:
Corrected

rcyl-structo commented on dtf/dtfserver.py:
```diff
@@ -25,19 +25,119 @@
+        self.plc.send_command("PLCClearSequenceComplete")
+        if self.plc.check_bit(self.plc.remove_bit, 0):
+            print("HomeVerticalStage Complete")
+            return dtfserver_pb2.GenericResponse(success=True, errorCode=0,  message="HomeVerticalStage complete")
```
correct formatting

siewqin-structo replied:
Corrected

rcyl-structo commented on dtf/dtfserver.py:
```diff
@@ -123,19 +218,237 @@ def SetGPIO_LightSource(self, request, context):
+                return dtfserver_pb2.InstructionResponse()
+
+        print("Platform Raised. Proceed to clear stage")
+        return dtfserver_pb2.InstructionResponse(success=True, errorCode=0,  message="Platform Raised", instruction="Ensure stage is cleared and door is closed before proceeding", pressNext="LowerPlatform")
```
correct formatting

siewqin-structo replied:
Corrected

rcyl-structo commented on dtf/dtfserver.py:
```diff
@@ -123,19 +218,237 @@ def SetGPIO_LightSource(self, request, context):
+                return dtfserver_pb2.InstructionResponse()
+
+        print("Platform Lowering Complete. Proceed to align")
+        return dtfserver_pb2.InstructionResponse(success=True, errorCode=0,  message="Platform Lowered", instruction="User may proceed to align. Ensure door is closed before proceeding.", pressNext="CompleteAlignment")
```
correct formatting

siewqin-structo replied:
Corrected

rcyl-structo commented on dtf/dtfserver.py:
```diff
@@ -123,19 +218,237 @@ def SetGPIO_LightSource(self, request, context):
+        self.plc.send_command("UITrigger2")
+        print("Platform Raising. Do not open door until operation is completed")
+
+        if (self.plc.poll_status_with_door(self.plc.remove_print_time_sec, self.plc.sequence_complete,1) is False):
```
correct formatting

siewqin-structo replied:
Corrected

rcyl-structo commented on dtf/dtfserver.py:
```diff
@@ -123,19 +218,237 @@ def SetGPIO_LightSource(self, request, context):
+
+        if self.plc.check_bit(self.plc.calib_valid_bit,1):
+            print("Calibration completed successfully")
+            return dtfserver_pb2.GenericResponse(success=True, errorCode=0,  message="Platform Alignment Complete")
```
correct formatting

siewqin-structo replied:
Corrected

rcyl-structo commented on dtf/dtfserver.py:
```diff
@@ -123,19 +218,237 @@ def SetGPIO_LightSource(self, request, context):
+            return dtfserver_pb2.GenericResponse(success=True, errorCode=0,  message="Platform Alignment Complete")
+        else:
+            self.handleError("Status Bit not 1", context, "CALIBVALID not 1")
+            return dtfserver_pb2.GenericResponse(success=False, errorCode=0,  message="Platform Alignment Failed")
```
correct formatting

siewqin-structo replied:
Corrected

rcyl-structo commented on dtf/dtfserver.py:
```diff
@@ -123,19 +218,237 @@ def SetGPIO_LightSource(self, request, context):
+        
+        if self.plc.check_bit(self.plc.calib_bit,1) and self.plc.check_bit(self.plc.ui_flag1_bit,1):
+            print("Platform is Raised. Proceed to clear stage")
+            return dtfserver_pb2.InstructionResponse(success=True, errorCode=0,  message="Platform is Raised", instruction="Ensure stage is cleared and door is closed before proceeding", pressNext="LowerPlatform")
```
correct formatting

siewqin-structo replied:
Corrected

rcyl-structo commented on dtf/dtfserver.py:
```diff
@@ -123,19 +218,237 @@ def SetGPIO_LightSource(self, request, context):
+
+        if self.plc.check_bit(self.plc.calib_bit,1) and self.plc.check_bit(self.plc.ui_flag2_bit,1):
+            print("Platform Lowering Complete. Proceed to align")
+            return dtfserver_pb2.InstructionResponse(success=True, errorCode=0,  message="Platform is Lowered", instruction="Align the platform and close the door before proceeding next step", pressNext="CompleteAlignment")
```
correct formatting

siewqin-structo replied:
Corrected

rcyl-structo commented on dtf/dtfserver.py:
```diff
@@ -123,19 +218,237 @@ def SetGPIO_LightSource(self, request, context):
+            self.handleError("Invalid Op", context, "This is implemented for AlignPlatform only")
+            return dtfserver_pb2.InstructionResponse()
+
+        if (self.plc.checkflags_with_door(self.plc.move_time_sec, self.plc.ui_flag1_bit, 1, self.plc.ui_flag2_bit, 1) is False):
```
correct formatting

siewqin-structo replied:
Corrected

rcyl-structo commented on dtf/dtfserver.py:
```diff
@@ -123,19 +218,237 @@ def SetGPIO_LightSource(self, request, context):
+
+        if self.plc.check_bit(self.plc.ui_flag1_bit,1):
+            print("Platform is Raised. Proceed to clear stage")
+            return dtfserver_pb2.InstructionResponse(success=True, errorCode=0,  message="Platform is Raised", instruction="Ensure stage is cleared and door is closed before proceeding", pressNext="LowerPlatform")
```
correct formatting

siewqin-structo replied:
Corrected

rcyl-structo commented on dtf/dtfserver.py:
```diff
@@ -123,19 +218,237 @@ def SetGPIO_LightSource(self, request, context):
+            return dtfserver_pb2.InstructionResponse(success=True, errorCode=0,  message="Platform is Raised", instruction="Ensure stage is cleared and door is closed before proceeding", pressNext="LowerPlatform")
+        elif self.plc.check_bit(self.plc.ui_flag2_bit,1):
+            print("Platform Lowering Complete. Proceed to align")
+            return dtfserver_pb2.InstructionResponse(success=True, errorCode=0,  message="Platform is Lowered", instruction="Align the platform and close the door before proceeding next step", pressNext="CompleteAlignment")
```
correct formatting

siewqin-structo replied:
Corrected

rcyl-structo commented on dtf/dtfserver.py:
```diff
@@ -123,19 +218,237 @@ def SetGPIO_LightSource(self, request, context):
+        if status:
+            print("TankClean Complete")
+            self.plc.reset()
+            return dtfserver_pb2.GenericResponse(success=status, errorCode=0,  message="TankClean Complete")
```
correct formatting

siewqin-structo replied:
Corrected

rcyl-structo commented on dtf/dtfserver.py:
```diff
@@ -123,19 +218,237 @@ def SetGPIO_LightSource(self, request, context):
+        cmd = "python3 startDMI.py &"
+        os.system(cmd)
+        print("Fan and LEDs are ON. Press CompleteInspection to end inspection")
+        return dtfserver_pb2.InstructionResponse(success=True, errorCode=0,  message="Fan and LEDs are ON", instruction="Ensure that the Structo GIF is flashing clearly on the LCD", pressNext="CompleteInspection to exit")
```
correct formatting

siewqin-structo replied:
Corrected

rcyl-structo commented on dtf/dtfserver.py:
```diff
@@ -123,19 +218,237 @@ def SetGPIO_LightSource(self, request, context):
+
+        if self.plc.check_bit(self.plc.light_src_bit,0) and self.plc.check_bit(self.plc.fan_bit,0):
+            print("LCDInspection Complete")
+            return dtfserver_pb2.GenericResponse(success=True, errorCode=0,  message="LCDInspection Complete")
```
correct formatting

siewqin-structo replied:
Corrected

rcyl-structo commented on dtf/dtfserver.py:
```diff
@@ -123,19 +218,237 @@ def SetGPIO_LightSource(self, request, context):
+            self.handleError("Status Bit not 1", context, "CALIBVALID not 1")
+            return dtfserver_pb2.GenericResponse(success=False, errorCode=0,  message="Platform Alignment Failed")
+
+    def DoorIsClosed(self, request, context):
```
can you add comments explaning the logic of this function?

rcyl-structo reviewed (commented):
Remove the generated files, dtfserver_pb2.py and dtfserver_pb2_grpc.py from the git repo. These are generated files and should not be part of the repo. 

In general, correct the formatting of some of the functions. You may use a autoformat tool in VS code to do this (i don't know which one, but sure plenty of them if you look it up)

siewqin-structo commented:
A summary of the changes in my code, on top of your comments:
1. Code Alignment/Formatting
2. Removed proto generated files
3. Standardized the use of send_command funtion
4. Adopted a rough state machine framework in attempt to neaten the codes under DTFAlignPlatform and HomeVerticalStage
5. Removed DoorIsClosed for both DTFAlignPlatform and HomeVerticalStage
---> Rationale: User can simply press the same function **_again_** to resume motion instead of using an additional function

rcyl-structo commented on dtf/dtfserver.py:
```diff
@@ -16,126 +16,555 @@
+            return dtfserver_pb2.InstructionResponse()
+
+        elif self.calibState == "CompleteAlignmentOngoing":
+            self.handleError("Door Open Terminate", context)
```
is this correct? are you missing some other code to check the door?

siewqin-structo replied:
No need to check the door anymore because during the final operation of raising the platform (completeAlignment), any opening of the door will terminate the entire calibration process.

rcyl-structo commented on dtf/dtfserver.py:
```diff
@@ -16,126 +16,555 @@
+                                                     pressNext=press)
+
+        elif self.calibState == "CompleteAlignmentOngoing":
+            self.handleError("Door Open Terminate", context)
```
is this correct? are you missing some other code to check the door?

rcyl-structo commented on dtf/dtfserver.py:
```diff
@@ -16,126 +16,555 @@
+            self.calibState = "CompleteAlignmentOngoing"
+
+        elif self.calibState == "CompleteAlignmentOngoing":
+            self.handleError("Door Open Terminate", context)
```
is this correct? are you missing some other code to check the door?

rcyl-structo commented on dtf/dtfserver.proto:
```diff
@@ -1,11 +1,12 @@
+    
     rpc DoPrintCycle(PrintingProfile) returns (GenericResponse) {}
     rpc DoExposeSlice(ExposureProfile) returns (GenericResponse) {}
-    rpc HomeVerticalStage(Empty) returns (GenericResponse) {}
```
Don't change the order of the functions, you might break existing code that calls these functions

rcyl-structo reviewed (changes requested):
Just maintain the order of the protos function
